### PR TITLE
MULE-10480 - Move deployment code to its own module

### PR DESCRIPTION
**Pull request by pablolagreca**

### CODE REVIEW

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/ApplicationDescriptorFactory.java [hunk 1]
```diff
@@ -104,18 +103,18 @@ public ApplicationDescriptor create(File artifactFolder) throws ArtifactDescript
   }
 
   private URL[] findLibraries(String appName) throws MalformedURLException {
-    return findJars(getAppLibFolder(appName)).toArray(new URL[0]);
+    return findJars(MuleFoldersUtil.getAppLibFolder(appName)).toArray(new URL[0]);
```
Comment (pablokraan) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/ApplicationDescriptorFactory.java]: Add static import

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/MuleDeploymentService.java [hunk 2]
```diff
@@ -74,7 +77,7 @@ public MuleDeploymentService(DefaultDomainFactory domainFactory, DefaultApplicat
     ArtifactDeployer<Domain> domainMuleDeployer = new DefaultArtifactDeployer<>();
 
     this.applicationDeployer = new DefaultArchiveDeployer<>(applicationMuleDeployer, applicationFactory, applications,
-                                                            NOP_ARTIFACT_DEPLOYMENT_TEMPLATE);
+                                                            ArtifactDeploymentTemplate.NOP_ARTIFACT_DEPLOYMENT_TEMPLATE);
```
Comment (pablokraan) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/MuleDeploymentService.java]: Add static import

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/MuleSharedDomainClassLoader.java [hunk 3]
```diff
@@ -35,7 +36,7 @@ public MuleSharedDomainClassLoader(String domain, ClassLoader parent, ClassLoade
   public URL findResource(String name) {
     URL resource = super.findResource(name);
     if (resource == null) {
-      File file = new File(getDomainFolder(getArtifactName()) + File.separator + name);
+      File file = new File(MuleFoldersUtil.getDomainFolder(getArtifactName()) + File.separator + name);
```
Comment (pablokraan) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/MuleSharedDomainClassLoader.java]: Add static import

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/domain/DefaultDomainFactory.java [hunk 4]
```diff
@@ -66,11 +67,11 @@ public Domain createArtifact(String artifactName) throws IOException {
   }
 
   private DomainDescriptor findDomain(String domainName) throws IOException {
-    if (DEFAULT_DOMAIN_NAME.equals(domainName)) {
-      return new EmptyDomainDescriptor(DEFAULT_DOMAIN_NAME);
+    if (Domain.DEFAULT_DOMAIN_NAME.equals(domainName)) {
```
Comment (pablokraan) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/domain/DefaultDomainFactory.java]: Add static import

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/domain/DefaultDomainFactory.java [hunk 5]
```diff
@@ -66,11 +67,11 @@ public Domain createArtifact(String artifactName) throws IOException {
...
+      return new EmptyDomainDescriptor(Domain.DEFAULT_DOMAIN_NAME);
     }
 
-    final File deploymentFile = getDeploymentFile(getDomainFolder(domainName));
+    final File deploymentFile = getDeploymentFile(MuleFoldersUtil.getDomainFolder(domainName));
```
Comment (pablokraan) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/domain/DefaultDomainFactory.java]: Add static import

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/domain/DomainClassLoaderFactory.java [hunk 6]
```diff
@@ -123,21 +123,21 @@ private ArtifactClassLoader getCustomDomainClassLoader(ClassLoaderLookupPolicy c
     try {
       List<URL> urls = new LinkedList<>();
       urls.add(MuleFoldersUtil.getDomainFolder(domain).toURI().toURL());
-      File domainLibraryFolder = getDomainLibFolder(domain);
+      File domainLibraryFolder = MuleFoldersUtil.getDomainLibFolder(domain);
```
Comment (pablokraan) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/domain/DomainClassLoaderFactory.java]: Add static import

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/domain/DomainStartedSplashScreen.java [hunk 7]
```diff
@@ -18,7 +22,7 @@
   @Override
   protected void createMessage(DomainDescriptor descriptor) {
     doBody(String.format("Started domain '%s'", descriptor.getName()));
-    if (RUNTIME_VERBOSE_PROPERTY.isEnabled()) {
+    if (SplashScreen.RUNTIME_VERBOSE_PROPERTY.isEnabled()) {
```
Comment (pablokraan) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/domain/DomainStartedSplashScreen.java]: Add static import

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java [hunk 8]
```diff
@@ -827,7 +830,7 @@ public void redeployModifiedDomainAndRedeployFailedApps() throws Exception {
 
     // change shared http config name to use a wrong name
     File domainConfigFile =
-        new File(domainsDir + "/" + sharedHttpBundleDomainFileBuilder.getDeployedPath(), DOMAIN_CONFIG_FILE_LOCATION);
+        new File(domainsDir + "/" + sharedHttpBundleDomainFileBuilder.getDeployedPath(), Domain.DOMAIN_CONFIG_FILE_LOCATION);
```
Comment (pablokraan) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java]: Add static import

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java [hunk 9]
```diff
@@ -2310,7 +2313,7 @@ public void refreshDomainClassloaderAfterRedeployment() throws Exception {
 
     // Ensure resources are registered at domain's registry
     Domain domain = findADomain(sharedHttpDomainFileBuilder.getId());
-    assertThat(domain.getMuleContext().getRegistry().get("http-listener-config"), not(is(nullValue())));
+    assertThat(domain.getMuleContext().getRegistry().get("http-listener-config"), not(is(IsNull.nullValue())));
```
Comment (pablokraan) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java]: Add static import

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/api/DeploymentService.java [hunk 10]
```diff
@@ -4,17 +4,17 @@
...
+package org.mule.runtime.module.deployment.api;
 
 import java.io.IOException;
 import java.net.URL;
 import java.util.Collection;
 import java.util.List;
 import java.util.concurrent.locks.ReentrantLock;
 
+import org.mule.runtime.module.deployment.api.application.Application;
+import org.mule.runtime.module.deployment.api.domain.Domain;
```
Comment (elrodro83) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/api/DeploymentService.java]: check the import order

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java [hunk 11]
```diff
@@ -2812,7 +2815,8 @@ private void assertMuleContextInitialized(final DeploymentListener listener, fin
 
       @Override
       public boolean test() {
-        verify(listener, times(1)).onMuleContextInitialised(eq(appName), any(MuleContext.class));
+        verify(listener, times(1)).onMuleContextInitialised(eq(appName),
+                                                            any(MuleContext.class));
```
Comment (pablokraan) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java]: There are many lines like this one that were changed with not reason, maybe you formatter config is different

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/application/ApplicationExtensionsManagerConfigurationBuilderTestCase.java [hunk 12]
```diff
@@ -78,7 +77,7 @@ public void before() throws Exception {
   @Test
   public void register() throws Exception {
     builder.doConfigure(muleContext);
-    ArgumentCaptor<ExtensionManifest> manifestCaptor = forClass(ExtensionManifest.class);
+    ArgumentCaptor<ExtensionManifest> manifestCaptor = ArgumentCaptor.forClass(ExtensionManifest.class);
```
Comment (pablokraan) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/application/ApplicationExtensionsManagerConfigurationBuilderTestCase.java]: Add static import

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/ApplicationDescriptorFactory.java (hunk 1 above)
Comment (elrodro83) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/ApplicationDescriptorFactory.java]: static imports

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/MuleDeploymentService.java (hunk 2 above)
Comment (elrodro83) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/MuleDeploymentService.java]: static import

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/MuleSharedDomainClassLoader.java (hunk 3 above)
Comment (elrodro83) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/MuleSharedDomainClassLoader.java]: static imports

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/application/ApplicationStartedSplashScreen.java [hunk 13]
```diff
@@ -4,23 +4,26 @@
...
-import org.mule.runtime.module.launcher.descriptor.ApplicationDescriptor;
-import org.mule.runtime.module.launcher.plugin.ArtifactPluginDescriptor;
+import org.mule.runtime.module.deployment.api.application.Application;
+import org.mule.runtime.module.deployment.internal.artifact.ArtifactStartedSplashScreen;
+import org.mule.runtime.module.deployment.internal.descriptor.ApplicationDescriptor;
+import org.mule.runtime.module.deployment.internal.plugin.ArtifactPluginDescriptor;
 
 import java.util.Set;
 
+import static java.lang.String.format;
```
Comment (elrodro83) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/application/ApplicationStartedSplashScreen.java]: import order

File: modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/descriptor/PropertiesDescriptorParser.java [hunk 14]
```diff
@@ -17,6 +16,9 @@
 
 import org.apache.commons.lang.BooleanUtils;
 
+import static org.mule.runtime.module.deployment.internal.descriptor.ApplicationDescriptor.DEFAULT_CONFIGURATION_RESOURCE;
+import static org.mule.runtime.module.reboot.MuleContainerBootstrapUtils.getMuleAppDir;
```
Comment (elrodro83) [inline modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/descriptor/PropertiesDescriptorParser.java]: import order

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java [hunk 15]
```diff
@@ -3198,7 +3204,8 @@ private void executeApplicationFlow(String flowName) throws MuleException {
         (Flow) deploymentService.getApplications().get(0).getMuleContext().getRegistry().lookupFlowConstruct(flowName);
     MuleMessage muleMessage = MuleMessage.builder().payload(TEST_MESSAGE).build();
 
-    mainFlow.process(MuleEvent.builder(DefaultMessageContext.create(mainFlow, TEST_CONNECTOR)).message(muleMessage)
+    mainFlow.process(MuleEvent.builder(DefaultMessageContext.create(mainFlow, AbstractMuleTestCase.TEST_CONNECTOR))
```
Comment (elrodro83) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java]: static import

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/application/DefaultMuleApplicationStatusTestCase.java [hunk 16]
```diff
@@ -53,7 +56,9 @@ public void initialState() {
   public void initialised() {
     // the context was initialised before we gave it to the application, so we need
     // to fire the notification again since the listener wasn't there
-    muleContext.fireNotification(new MuleContextNotification(muleContext, MuleContextNotification.CONTEXT_INITIALISED));
+    muleContext
+        .fireNotification(new MuleContextNotification(muleContext,
+                                                      MuleContextNotification.CONTEXT_INITIALISED));
```
Comment (elrodro83) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/application/DefaultMuleApplicationStatusTestCase.java]: static import

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/domain/DomainClassLoaderFactoryTestCase.java [hunk 17]
```diff
@@ -49,8 +50,9 @@ public void createClassLoaderUsingDefaultDomain() {
 
     DomainDescriptor descriptor = getTestDescriptor(DEFAULT_DOMAIN_NAME);
 
-    assertThat(new DomainClassLoaderFactory(getClass().getClassLoader()).create(containerClassLoader, descriptor, emptyList())
-        .getArtifactName(), is(DEFAULT_DOMAIN_NAME));
+    assertThat(new DomainClassLoaderFactory(getClass().getClassLoader())
+        .create(containerClassLoader, descriptor, emptyList())
+        .getArtifactName(), CoreMatchers.is(DEFAULT_DOMAIN_NAME));
```
Comment (elrodro83) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/domain/DomainClassLoaderFactoryTestCase.java]: static import

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/domain/DomainStartedSplashScreenTestCase.java [hunk 18]
```diff
@@ -4,29 +4,31 @@
...
 
 public class DomainStartedSplashScreenTestCase extends AbstractSplashScreenTestCase<DomainStartedSplashScreen> {
 
   private static final String DOMAIN_NAME = "simpleDomain";
-  private static final String DOMAIN_LIB_PATH = String.format("domains/%s/lib", DOMAIN_NAME);
+  private static final String DOMAIN_LIB_PATH = format("domains/%s/lib", DOMAIN_NAME);
   private static final String SOME_JAR = "some.jar";
   private static final String MY_ZIP = "myZip.zip";
-  private DomainDescriptor descriptor = mock(DomainDescriptor.class);
+  private DomainDescriptor descriptor = Mockito.mock(DomainDescriptor.class);
```
Comment (elrodro83) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/domain/DomainStartedSplashScreenTestCase.java]: static import

File: modules/launcher/src/test/java/org/mule/runtime/module/launcher/MuleContainerStartupSplashScreenTestCase.java [hunk 19]
```diff
@@ -27,7 +29,7 @@
 
   @BeforeClass
   public static void setUpPatches() {
-    File libFolder = newFile(workingDirectory.getRoot(), "lib/user");
+    File libFolder = newFile(AbstractSplashScreenTestCase.workingDirectory.getRoot(), "lib/user");
```
Comment (elrodro83) [inline modules/launcher/src/test/java/org/mule/runtime/module/launcher/MuleContainerStartupSplashScreenTestCase.java]: static import

**Pull request merged**